shotover-proxy/src/message/mod.rs

```diff
@@ -60,6 +61,8 @@ pub struct Message {
     // TODO: Not a fan of this field and we could get rid of it by making TimestampTagger an implicit part of ConsistentScatter
     // This metadata field is only used for communication between transforms and should not be touched by sinks or sources
     pub meta_timestamp: Option<i64>,
+
+    pub compression: Option<Compression>,
```

Comment:
Making this an Option is duplicating the Compression::None state.

Reply:
It's to indicate that the message has no compression set. e.g. a redis message has no concept of compression, whereas a Cassandra messages does. I'd think of it like compression = null rather than compression = compression::none.

Reply:
If None indicates a non-cassandra message then I think thats a reasonable attempt at future proofing our logic, but we should make that clear by making the field `pub codec_state: CodecState` and having a:
```rust
enum CodecState {
    Cassandra { compression: Compression },
    Redis,
}
```